index.js

```diff
@@ -1,74 +1,278 @@
-var analyze = require('./lib/analyze-raptor');
-var Promise = require('promise');
-var fs = require('fs');
-var debug = require('debug')('raptor-regression');
-
-var metric = 'visuallyLoaded';
-
-var readFile = function(dataFile) {
-  debug('Processing %s', dataFile);
-  return new Promise(function(resolve, reject) {
-    fs.readFile(dataFile, { encoding: 'utf8' }, function(err, content) {
-      if (err) {
-        return reject(err);
-      }
-
-      var data = JSON.parse(content.split('\n'));
-      resolve(data);
-    });
-  });
+var R = require('ramda');
+
+var BACKWINDOW_SIZE = 12;
+var FOREWINDOW_SIZE = 12;
+var T_THRESHOLD = 7;
+
+/**
+ * Always get 1
+ * @returns 1
+ */
+var single = R.always(1);
+
+/**
+ * Always get zero
+ * @returns 0
+ */
+var zero = R.always(0);
+
+/**
+ * Always get Infinity
+ * @returns Infinity
+ */
+var infinity = R.always(Infinity);
+
+/**
+ * Get the square of a number
+ * @param {Number}
+ * @returns {Number}
+ */
+var square = R.partialRight(Math.pow, 2);
+
+/**
+ * Square every number in a collection
+ * @param {Array<Number>} numbers
+ * @returns {Array<Number>}
+ */
+var squareAll = R.map(square);
+
+/**
+ * Get the numerator for a variance calculation by summing the squares of a collection
+ * @param {Array<Number>} numbers
+ * @returns {Number}
+ */
+var varianceNumerator = R.pipe(squareAll, R.sum);
+
+/**
+ * Get the denominator for a variance calculation by decrementing the length of a collection
+ * @param {Array} array
+ * @returns {Number}
+ */
+var varianceDivisor = R.pipe(R.length, R.dec);
```

Comment:
So the collection that R.length is using comes from R.converge on line 74, correct? I'm so used to something like R.length(arrayArg)